L1Trigger/L1TGlobal/interface/GlobalBoard.h

```diff
@@ -164,6 +165,32 @@ namespace l1t {
     /// pointer to Tau data list
     inline const BXVector<const GlobalExtBlk*>* getCandL1External() const { return m_candL1External; }
 
+    //initializer prescale counter using a semi-random value between [1, prescale value]
+    static const std::vector<double> semirandomNumber(const edm::Event& iEvent,
+                                                      const std::vector<double>& prescaleFactorsAlgoTrig) {
+      std::vector<double> out(prescaleFactorsAlgoTrig.size(), 1.);
+      //pick a random number from a combination of run, lumi, event numbers (different number for different threads)
+      std::srand(0);
+      std::srand(std::rand() + iEvent.id().run());
+      std::srand(std::rand() + iEvent.id().luminosityBlock());
+      //this causes different semirandomNumber number for different threads
+      std::srand(std::rand() + iEvent.id().event());  //reminder: different threads have different initial event number
+      const double semirandom = std::rand();
+      for (size_t i = 0; i < prescaleFactorsAlgoTrig.size(); i++) {
+        const double ps = prescaleFactorsAlgoTrig.at(i);
+        if (ps == 0 || ps == 1) {  //do not touch ps = 0 and ps = 1
+          out[i] = ps;
+        } else {  //replace ps with a semirandom number between [1,ps]
+          out[i] = semirandom - floor(semirandom / ps) * ps;
+          if (out[i] == 0)
+            out[i] = ps;
+          assert(out[i] > 0);
+          assert(out[i] <= ps);
+        }
+      }
+      return out;
+    }
+
```

Comment:
It looks like my reply is lost.

- I replaced the assert with a warning. In case of problems it will simply use the standard method ( take PS )

- I think that it should work also with fractional prescales, as they don't have a specific treatment in GlobalBoard.cc (not 100% sure though)

-  `rand` should be common to among all the algos in order to preserve the synchronization of the L1 prescales (eg. L1_ZeroBias1 with PS = 20 should be always in the shadow of L1_ZeroBias2 with PS =10 )

- thanks, I partially copied your code. I had to make some change to preserve the behavior described above.